centml/compiler/backend.py

```diff
@@ -50,20 +53,21 @@ def module(self):
     def inputs(self):
         self._inputs = None
 
-    def _get_model_id(self, flow_graph: hidet.FlowGraph) -> str:
-        if not flow_graph:
-            raise Exception("Getting model id: flow graph is None.")
-
-        with tempfile.NamedTemporaryFile() as temp_file:
-            try:
-                hidet.save_graph(flow_graph, temp_file.name)
-            except Exception as e:
-                raise Exception(f"Getting model id: failed to save FlowGraph. {e}\n") from e
+    def _get_model_id(self) -> str:
+        # The GraphModule can be large, so lets serialize it to disk
+        # This saves a zip file full of pickled files.
+        try:
+            torch.save(self.module, self.serialized_model_path, pickle_protocol=config_instance.PICKLE_PROTOCOL)
+        except Exception as e:
+            raise Exception(f"Failed to save module with torch.save: {e}") from e
 
-            with open(temp_file.name, "rb") as f:
-                flow_graph_hash = hashlib.md5(f.read()).hexdigest()
+        sha_hash = hashlib.sha256()
+        with open(self.serialized_model_path, "rb") as serialized_model_file:
+            # Read in chunks to not load too much into memory
+            for block in iter(lambda: serialized_model_file.read(4096), b""):
```

Comment:
can you set this 4096 as a constant variable somewhere? may be in the config class. But no need to make it configurable. it can be hard coded as 4096

Reply:
yes, I'll define it in the config class